lib/core/context.c

```diff
@@ -190,3 +190,28 @@ void h2o_context_update_timestamp_string_cache(h2o_context_t *ctx)
     h2o_time2str_rfc1123(ctx->_timestamp_cache.value->rfc1123, &gmt);
     h2o_time2str_log(ctx->_timestamp_cache.value->log, ctx->_timestamp_cache.tv_at.tv_sec);
 }
+
+int h2o_context_close_idle_connections(h2o_context_t *ctx, int max_connections_to_close, int min_age)
+{
+    if (max_connections_to_close <= 0)
+        return 0;
+
+    int closed = 0;
+    h2o_linklist_t *conn_list[] = {
+        &ctx->_idle_conns,
+    };
+    H2O_CONN_LIST_FOREACH(h2o_conn_t * conn, conn_list, {
+        struct timeval now = h2o_gettimeofday(ctx->loop);
+        if (now.tv_sec - conn->connected_at.tv_sec < min_age)
+            continue;
+        if (conn->callbacks->close_idle_connection == NULL) {
```

Comment:
Assuming that we would require `h2o_conn_t`'s to define this callback, I do not think we need to test and print an error. Our custom is to let the program die when we detect a violation of design contract, under the belief that doing so helps us maintain the code in good shape.